content/blog/2022/2022-01-18-getting-started-with-spring-boot.md

```diff
@@ -0,0 +1,915 @@
+Let us see how to define the file-based H2 database:
+
+```text
+spring.datasource.url=jdbc:h2:file:/Users/mateostjepanovic/Documents/git/code-examples/spring-boot/begginers-guide/src/main/resources/data/demo;AUTO_SERVER=true
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+spring.datasource.driver-class-name=org.h2.Driver
+# Rest of the configuration is omitted
+```
+After defining the `jdbc` and the `h2` keywords, we note that we want to use a file-based H2 database with the `file` keyword. The last part of the URL is the absolute path to the folder where we want to save our database.
+
+{{% info title="The H2 database" %}}
+Please note that we should use the H2 only for the development phase. When going to the production environment, move to something more production-ready ( Oracle, PostgreSQL, etc.).
+{{% /info %}}
+
+#### Defining the Database Login Information
+After we defined the URL for the database we need the login informations for that database:
+
+```text
+spring.datasource.username=username
+spring.datasource.password=password
+spring.h2.console.enabled=true
+
+# Rest of the configuration is omitted
+```
+We define the username and password for the database.
+
+Please define that the password and username should be externalized and encrypted in the production environment.
+
+More about externalizing configurations can be found in the [Build Once, Run Anywhere: Externalize Your Configuration](https://reflectoring.io/externalize-configuration/) article.
+
+#### Defining Schema Creation
+We can set the configuration to autocreate the schema for us:
+```text
+spring.jpa.hibernate.ddl-auto=create
+spring.jpa.generate-ddl=true
+
+# Rest of the configuration is omitted
+```
+The Hibernate DDL is used to generate the schema using the entity definitions. With the `ddl-auto`, we set that we want to destroy and recreate the schema on each run.
+
+## Building the Data Access Layer
+
+{{% image alt="Spring Boot High-level arch " src="images/posts/spring-boot-begginer-guide/SpringApplication-dataAccessLayer.png" %}}
+
+After creating entities, we can develop the data access layer. The data access layer allows us to use methods to manipulate the data in the database. We will build the data access layer using the repository pattern.
+
+The repository pattern is the design pattern that leverages the usage of interfaces for connection to the database. The repository interface hides all implementation details of connecting to the database, maintaining the connection, transactions, etc.
+
+We will show how to create the repository for the entity and explain how we use it.
+
+### Creating the Book Repository
+```java
+public interface BookRepository extends JpaRepository<Book, Long> {
+   // Rest of the code is omitted 
+}
+```
+Extending `JpaRepository` turns our interface into the repository bean for the application context to control. 
```

Comment:
What is the application context? This is the first time this word is used. As a newbie, I'm confused. Maybe add a short section about the Spring programming model after the "Setting up the project" section? That section should explain the application context, what a bean is, what autowiring is and how it works. Maybe  link to https://reflectoring.io/constructor-injection/

Just realized that you have a section like this in "Injecting the Repository Class". Good! Mention that the application context is explained in that section, then!